apps/files/src/helpers/collaborators.js

```diff
@@ -0,0 +1,74 @@
+import roles from './collaboratorRoles'
+
+/**
+ * Bitmask of ownCloud permission
+ */
+export const permissionsBitmask = {
+  read: 1,
+  update: 2,
+  create: 4,
+  delete: 8,
+  share: 16,
+  all: 31
+}
+
+/**
+ * Maps role to its bitmask
+ * @param {object} role  Role which is to be mapped to permissions
+ * @param {array} additionalPermissions  Array of selected additional permissions
+ * @returns {number}  Role bitmask
+ */
+export function roleToBitmask (role, additionalPermissions = []) {
+  let bitmask = 0
+
+  for (const permission of role.permissions) {
+    bitmask += permissionsBitmask[permission]
+  }
+
+  if (additionalPermissions) {
+    for (const additionalPermission of additionalPermissions) {
+      bitmask += permissionsBitmask[additionalPermission]
+    }
+  }
+
+  return bitmask
+}
+
+/**
+ * Maps bitmask to role
+ * @param {number} bitmask Permissions which are to be mapped to role
+ * @param {boolean} isFolder Defines if the item is folder
+ * @returns {object} Role mapped to the bitmask
+ */
+export function bitmaskToRole (bitmask, isFolder) {
+  const currentRoles = roles({ isFolder: isFolder })
+  bitmask = parseInt(bitmask)
+
+  for (const role in currentRoles) {
+    const currentRole = currentRoles[role]
+    let rolePermissionsBitmask = 0
+
+    // Get bitmask of basic permissions
+    currentRole.permissions.forEach(permission => {
+      rolePermissionsBitmask += permissionsBitmask[permission]
+    })
+
+    // Copy basic permissions bitmask so that in case that no additional permissions have been set, the role is still mapped
+    const roleBasicPermissionsBitmask = rolePermissionsBitmask
+
+    // Get bitmask of additional permissions
+    const additionalPermissions = currentRole.additionalPermissions
+    for (const additionalPermission in additionalPermissions) {
+      rolePermissionsBitmask += permissionsBitmask[additionalPermissions[additionalPermission].name]
+    }
+
+    if (
+      rolePermissionsBitmask === bitmask ||
```

Comment:
optional: might be doable using the `&` for bit masking/matching

=> or add todo for later